astropy/table/table.py

```diff
@@ -3631,6 +3631,7 @@ def _is_mixin_for_table(self, col):
         return has_info_class(col, MixinInfo)
 
     def _convert_col_for_table(self, col):
+        from astropy.units import Unit
```

Comment:
You can just add `Unit` to the line `from astropy.units import Quantity, QuantityInfo` at the top.